assets/dev/js/editor/atomic-widgets/views/a-container.js

```diff
@@ -0,0 +1,347 @@
+		};
+	},
+
+	onDeviceModeChange() {
+		SetDirectionMode.set( this.getContainer() );
+	},
+
+	getDirectionSettingKey() {
+		const containerType = this.container.settings.get( 'container_type' ),
+			directionSettingKey = 'grid' === containerType
+				? 'grid_auto_flow'
+				: 'flex_direction';
+
+		return directionSettingKey;
+	},
+
+	behaviors() {
+		const behaviors = BaseElementView.prototype.behaviors.apply( this, arguments );
+
+		_.extend( behaviors, {
+			// TODO: Remove. It's a temporary solution for the Navigator sortable.
+			Sortable: {
+				behaviorClass: require( 'elementor-behaviors/sortable' ),
+				elChildType: 'widget',
+			},
+			Resizable: {
+				behaviorClass: WidgetResizable,
+			},
+		} );
+
+		return elementor.hooks.applyFilters( 'elements/container/behaviors', behaviors, this );
+	},
+
+	onDestroy() {
+		BaseElementView.prototype.onDestroy.apply( this, arguments );
+
+		elementor.stopListening( elementor.channels.deviceMode, 'change', this.onDeviceModeChange );
+	},
+
+	/**
+	 * TODO: Remove. It's a temporary solution for the Navigator sortable.
+	 *
+	 * @return {{}} options
+	 */
+	getSortableOptions() {
+		// TODO: Temporary hack.
+		return {
+			preventInit: true,
+		};
+	},
+
+	getDroppableAxis() {
+		const isColumnDefault = ( ContainerHelper.DIRECTION_DEFAULT === ContainerHelper.DIRECTION_COLUMN ),
+			currentDirection = this.getContainer().settings.get( this.getDirectionSettingKey() );
+
+		const axisMap = {
+			[ ContainerHelper.DIRECTION_COLUMN ]: 'vertical',
+			[ ContainerHelper.DIRECTION_COLUMN_REVERSED ]: 'vertical',
+			[ ContainerHelper.DIRECTION_ROW ]: 'horizontal',
+			[ ContainerHelper.DIRECTION_ROW_REVERSED ]: 'horizontal',
+			'': isColumnDefault ? 'vertical' : 'horizontal',
+		};
+
+		return axisMap[ currentDirection ];
+	},
+
+	getDroppableOptions() {
+		const items = this.isBoxedWidth()
+			? '> .elementor-widget, > .e-con-full, > .e-con > .e-con-inner, > .elementor-empty-view > .elementor-first-add'
+			: '> .elementor-element, > .elementor-empty-view .elementor-first-add';
+
+		return {
+			axis: this.getDroppableAxis(),
+			items,
+			groups: [ 'elementor-element' ],
+			horizontalThreshold: 5, // TODO: Stop the magic.
+			isDroppingAllowed: this.isDroppingAllowed.bind( this ),
+			currentElementClass: 'elementor-html5dnd-current-element',
+			placeholderClass: 'elementor-sortable-placeholder elementor-widget-placeholder',
+			hasDraggingOnChildClass: 'e-dragging-over',
+			getDropContainer: () => this.getContainer(),
+			onDropping: ( side, event ) => {
+				event.stopPropagation();
+
+				// Triggering drag end manually, since it won't fired above iframe
+				elementor.getPreviewView().onPanelElementDragEnd();
+
+				const draggedView = elementor.channels.editor.request( 'element:dragged' ),
+					draggingInSameParent = ( draggedView?.parent === this ),
+					hasInnerContainer = jQuery( event.currentTarget ).hasClass( 'e-con-inner' ),
```

Comment:
`hasInnerContainer` - related to flexbox on V1

Reply:
Removed